### PR TITLE
<TreeView /> - External state management possibilities verification

### PR DESCRIPTION
**Pull request by liorzisman**

### CODE REVIEW

File: test/components/tree-view.spec.tsx [hunk 1]
```diff
@@ -47,6 +48,45 @@ const treeData: TreeItemData[] = [
     }
 ];
 
+const newTreeData = JSON.parse(JSON.stringify(treeData));
```
Comment (danielstro) [inline test/components/tree-view.spec.tsx]: I assume that this is meant to copy treeData. Use newTreeData = {...treeData}, it's much clearer.

File: test/components/tree-view.spec.tsx [hunk 2]
```diff
@@ -47,6 +48,45 @@ const treeData: TreeItemData[] = [
...
+newTreeData[0].children![2].children!.push({label: 'Kaiserschmarrn'});
+
+export interface TreeViewWrapperState {
+    treeData: object[];
+}
+
+export class TreeViewWrapper extends React.Component<{}, TreeViewWrapperState> {
+
+    constructor() {
```
Comment (danielstro) [inline test/components/tree-view.spec.tsx]: You don't need to call the constructor here, just use: 
public state = {treeData};

File: test/components/tree-view.spec.tsx (hunk 1 above)
Comment (liorzisman) [reply test/components/tree-view.spec.tsx]: that would be shallow copying

File: test/components/tree-view.spec.tsx (hunk 2 above)
Comment (liorzisman) [reply test/components/tree-view.spec.tsx]: done

File: test/components/tree-view.spec.tsx (hunk 1 above)
Comment (danielstro) [reply test/components/tree-view.spec.tsx]: Well what do you know
https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Deep_Clone

Comment (JoMarton) [reply test/components/tree-view.spec.tsx]: please add a comment explaining this

File: test/components/tree-view.spec.tsx [hunk 3]
```diff
@@ -378,6 +412,40 @@ describe('<TreeView />', () => {
             });
         });
 
+        describe('Reaction to dataSource changes', () => {
+            it('renders the additional item when a new data array is passed', async () => {
+                const {select, waitForDom, result} = clientRenderer.render(<TreeViewWrapper />);
+
+                expandItemWithLabel(select, treeData[0].label);
```
Comment (JoMarton) [inline test/components/tree-view.spec.tsx]: this test should be referencing `newTreeData` right?

File: test/components/tree-view.spec.tsx (hunk 1 above)
Comment (JoMarton) [reply test/components/tree-view.spec.tsx]: also please rename this constant to be clear that it has no real relation to `treeData`

**Pull request merged**